multichain-aggregator/multichain-aggregator-server/src/services/multichain_aggregator.rs

```diff
@@ -58,19 +74,75 @@ impl MultichainAggregatorService for MultichainAggregator {
         }))
     }
 
+    async fn list_addresses(
+        &self,
+        request: Request<ListAddressesRequest>,
+    ) -> Result<Response<ListAddressesResponse>, Status> {
+        let inner = request.into_inner();
+
+        let page_token: Option<(alloy_primitives::Address, logic::ChainId)> =
+            inner.page_token.map(parse_query_2).transpose()?;
+        let page_size = self.normalize_page_size(inner.page_size);
+
+        let (addresses, next_page_token) = logic::repository::addresses::search_by_query_paginated(
```

Comment:
Nit: it confuses me a bit that 'search' is called within 'list'. So, the endpoint seems more like a search to me 🤔